nuxt.config.js

```diff
@@ -49,62 +69,86 @@ module.exports = {
         ],
     },
     loading: { color: '#00BC87', height: '3px' },
-    router: {
-        middleware: ['modules'],
-    },
-    css: NUXT_CONFIG.css,
-    plugins: NUXT_CONFIG.plugins,
-    styleResources: NUXT_CONFIG.styleResources,
-    modules: [
+    modulesDir: ['node_modules', 'modules'],
+    buildModules: [
+        '~/modules/helpers/modulesLoader',
         '@nuxtjs/router',
+        ...MODULES_TO_LOAD,
+        '~/modules/helpers/symlinksCreator',
+        '~/modules/helpers/registerRouter',
+        '~/modules/helpers/registerExtends',
+        '~/modules/helpers/registerMiddleware',
+        '~/modules/helpers/registerStore',
+    ],
+    modules: [
         '@nuxtjs/axios',
-        'cookie-universal-nuxt',
         '@nuxtjs/style-resources',
-        [
-            '@nuxtjs/component-cache',
-            {
-                maxAge: 1000 * 60 * 60,
-            },
-        ],
+        ['@nuxtjs/component-cache', { maxAge: 1000 * 60 * 60 }],
+        'cookie-universal-nuxt',
     ],
+    modulesLoader: {
+        activeModules: ACTIVE_MODULES,
+        inactiveModules: INACTIVE_MODULES,
+        requiredModules: REQUIRED_MODULES,
+        isDev: IS_DEV,
+    },
+    router: {
+        middleware: ['modulesMiddlewareLoader'],
+    },
+    symlinksCreator: {
+        modulesDir: NPM_MODULES,
+    },
+    registerRouter: {
+        modules: ACTIVE_MODULES,
+    },
+    registerExtends: {
+        modules: ACTIVE_MODULES,
+    },
+    registerMiddleware: {
+        modules: ACTIVE_MODULES,
+    },
+    registerStore: {
+        modules: ACTIVE_MODULES,
+    },
     axios: {
         baseURL: BASE_URL || 'http://localhost:8000',
     },
     build: {
-        // terser: {
-        //     parallel: true,
-        //     sourceMap: true,
-        //     exclude: /\/store/,
-        //     extractComments: {
-        //         filename: 'LICENSES',
-        //     },
-        //     terserOptions: {
-        //         output: {
-        //             comments: /^\**!|@preserve|@license|@cc_on/,
-        //         },
-        //     },
-        // },
+        terser: {
+            parallel: true,
+            sourceMap: true,
+            // exclude: /\/store/,
```

Comment:
comments

Reply:
All comments in this file are needed for later webpack configuration.